### PR TITLE
Updates to Anonymous Landing Page

### LINKED ISSUE #2681: Re-organize the top and bottom bars on the landing page for anonymous users
## Motivation

* The search button is too hidden on the anonymous landing page (some users have difficulties exploring the projects/datasets) and makes it difficult to explore content.
* We have a bunch of links on the top bar pointing to social media that we don't keep updated. We might use the space for more relevant pages.

## Proposal

- Move the Search button to a more prominent position.
- Re-arrange the links on the top bar and bottom bar, potentially moving social icons to its own section.
- Logged users: focus the search input box immediately when reaching the search page (UI only)

### PR DESCRIPTION
**Pull request by ciyer**
An update to the landing page.

Fix #2681

/deploy renku=renku-ui/3.11-acceptance-tests  #persist

### CODE REVIEW

**Comment by ciyer** [comment]
From the meeting today:
- Buttons on first section should have translucent background with white text and outlines that turn green on hover
- "Get Started" links to https://renku.readthedocs.io/en/stable/tutorials/01_firststeps.html
- Leave the "Connecting dots" section unchanged
- Drop the Explore column in the bottom nav
- "Renku version" links to the the same page as the release number
- "Why Renku" links to https://renku.readthedocs.io/en/stable/introduction/why.html
- Remove "Features page" from the Help page

**Comment by ciyer** [comment]
Testing: check that the new design matches [Figma](https://www.figma.com/file/qFKV4jeWVr0v6y9XB0bmQO/Make-the-search-button-more-immediate-%232681?type=design&node-id=358%3A3472&mode=design&t=54yP7xhG6NN6H0jM-1)

File: client/src/landing/anonymousHomeNav.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                title="Documentation"
+                url={Docs.READ_THE_DOCS_ROOT}
+              />
+              <BottomNavExternalLink title="Get Started" url={tutorialLink} />
+              <BottomNavLink title="Help" to={Url.get(Url.pages.help)} />
+            </BottomNavSection>
+          </Col>
+          <Col md={3}>
+            <BottomNavSection sectionTitle="Community">
+              <BottomNavExternalLink title="Form" url={Links.DISCOURSE} />
```
Comment (lorenzo-cavazzi) [inline client/src/landing/anonymousHomeNav.tsx:106]:
```suggestion
              <BottomNavExternalLink title="Forum" url={Links.DISCOURSE} />
```

File: client/src/landing/anonymousHomeNav.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+              <BottomNavExternalLink title="Chat (Gitter)" url={Links.GITTER} />
+              <BottomNavExternalLink title="GitHub" url={Links.GITHUB} />
+            </BottomNavSection>
+          </Col>
+          <Col md={3}>
+            <BottomNavSection sectionTitle="Follow us">
+              <BottomNavExternalLink title="Twitter" url={Links.TWITTER} />
```
Comment (leafty) [inline client/src/landing/anonymousHomeNav.tsx:113]: :nerd_face: 
```suggestion
              <BottomNavExternalLink title="𝕏 (formerly Twitter)" url={Links.TWITTER} />
```

File: client/src/landing/anonymousHomeNav.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+                <TopNavExternalLink
+                  title="Renku CLI Docs"
+                  url={RenkuPythonDocs.READ_THE_DOCS_ROOT}
+                />
+              </NavItem>
+              <NavItem>
+                <hr className="dropdown-divider mb-2" />
+              </NavItem>
+              <NavItem className="nav-item mb-2">
+                <TopNavExternalLink title="Form" url={Links.DISCOURSE} />
```
Comment (lorenzo-cavazzi) [inline client/src/landing/anonymousHomeNav.tsx:214]:
```suggestion
                <TopNavExternalLink title="Forum" url={Links.DISCOURSE} />
```

File: client/src/styles/bootstrap_ext/_button.scss [hunk 4]
```diff
@@ -135,6 +135,18 @@ $borderRadius: 1000px;
   }
 }
 
+.btn-outline-rk-white {
```
Comment (leafty) [inline client/src/styles/bootstrap_ext/_button.scss:138]: This class is generated by Bootstrap, I would prefer if we do not change its behavior globally. Would it be possible to override this in a (S)CSS module?

Also, overriding the color can be done with the bootstrap CSS variables (e.g. check the overrides from `btn-danger` or `btn-outline-danger`).

File: client/src/styles/components/_renku_home.scss [hunk 5]
```diff
@@ -7,10 +7,24 @@ header.rk-anon-home .nav-link {
   }
 }
 
-nav.rk-anon-home .nav-link {
+.nav-item hr.dropdown-divider {
+  @extend .my-2, .mx-3;
```
Comment (leafty) [inline client/src/styles/components/_renku_home.scss:11]: It seems simpler to assign `my-2` and `mx-3` to the class name.

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
This looks good! A couple of tiny changes before merging:

- The buttons margin/spacing could be improved by removing the right margin and adding X margins, and adding either the bottom or Y margins. Currently, they have no bottom margin and the first 2 buttons have a right margin, making their borders collapse together on smaller screens and moving them into a non horizonatally centered position.
image

- Bottom margins on the top container are inconsistent, depending on the browser size
Peek 2023-08-22 13-23

- See inline typos

File: client/src/styles/components/_renku_home.scss [hunk 6]
```diff
@@ -100,6 +114,17 @@ nav.rk-anon-home .nav-link {
   }
 }
 
+#rk-anon-home-bottom-nav {
+  @extend .section-content, .bg-primary;
+  color: $rk-white;
```
Comment (leafty) [inline client/src/styles/components/_renku_home.scss:119]: Use `text-rk-white` as a class name.

File: client/src/styles/components/_renku_home.scss [hunk 7]
```diff
@@ -100,6 +114,17 @@ nav.rk-anon-home .nav-link {
...
+  min-height: 300px;
+}
+
+ul.rk-anon-home-bottom-nav-section {
+  list-style-type: none;
```
Comment (leafty) [inline client/src/styles/components/_renku_home.scss:124]: `list-unstyled`: https://getbootstrap.com/docs/5.3/content/typography/#unstyled

File: client/src/styles/components/_renku_home.scss [hunk 8]
```diff
@@ -100,6 +114,17 @@ nav.rk-anon-home .nav-link {
...
+  padding-left: 0px;
```
Comment (leafty) [inline client/src/styles/components/_renku_home.scss:125]: `pl-0`

**Review by leafty — COMMENTED** [review]
About styling: I would suggest avoiding global CSS definitions and using Bootstrap's utility classes as much as possible. For cases when custom styling is required, (S)CSS modules seem appropriate as they will not affect components outside of their scope.

File: client/src/styles/components/_renku_home.scss (hunk 5 above)
Comment (ciyer) [reply client/src/styles/components/_renku_home.scss:11]: Good point.

**Comment by ciyer** [comment]
Thanks for the reviews, it's important that this page be polished. I think I have addressed all requests.

File: client/src/landing/anonymousHome.types.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    custom: {
+      enabled: boolean;
+      main: {
+        backgroundImage: {
+          url: string;
+        };
+        contentMd: string;
+      };
+    };
+    projects: [];
```
Comment (leafty) [inline client/src/landing/anonymousHome.types.ts:31]: Unless `never[]` is intended:

```suggestion
    projects: unknown[];
```

File: client/src/landing/anonymousHomeNav.tsx (hunk 2 above)
Comment (leafty) [reply client/src/landing/anonymousHomeNav.tsx:113]: Some companies have already updated it actually. e.g. a marketing email I received:
image

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm.
Keep using "Twitter" works fine for me since the new name is hard to relate to the product without the icon or specifying that it's actually what used to be Twitter

**Pull request merged**